js/digifinex.js

```diff
@@ -0,0 +1,515 @@
+            'filled': this.safeFloat (order, 'executed_amount'),
+            'status': status,
+            'fee': undefined,
+        };
+        return result;
+    }
+
+    async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        // exchange-specific non-unified parameters: page, type
+        // page: Page Num, page=1 for 1st page. None for 1st page by default.
+        // type：buy/sell/buy_market/sell_market，none for all types
+        const markets = await this.loadMarkets ();
+        const market = markets[symbol];
+        symbol = market['id'];
+        const request = {
+            'symbol': symbol,
+        };
+        const response = await this.privateGetOpenOrders (this.extend (request, params));
+        const orders = this.parseOrders (response['orders'], market, since, limit);
+        return orders;
+    }
+
+    async fetchClosedOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        // Only supports historical orders of last 3 days
+        // `since` has only date precision in digifinex
+        // exchange-specific non-unified parameters: page, type
+        // page: Page Num, page=1 for 1st page. None for 1st page by default.
+        // type：buy/sell/buy_market/sell_market，none for all types
+        const markets = await this.loadMarkets ();
+        const market = markets[symbol];
+        symbol = market['id'];
+        const request = {
+            'symbol': symbol,
+        };
+        if (since) {
+            request['date'] = this.dateUTC8 (since);
+        }
+        const response = await this.privateGetOrderHistory (params);
+        const filtered = this.parseOrders (response['orders'], market, since, limit);
+        return filtered;
+    }
+
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        const response = await this.privateGetOrderInfo (this.extend ({
+            'order_id': parseInt (id),
+        }, params));
+        return this.parseOrder (response);
+    }
+
+    parseOHLCV (ohlcv, market = undefined, timeframe = '1m', since = undefined, limit = undefined) {
+        return [
+            ohlcv[0] * 1000,
+            ohlcv[5],
+            ohlcv[3],
+            ohlcv[4],
+            ohlcv[2],
+            ohlcv[1],
+        ];
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        const markets = await this.loadMarkets ();
+        const market = markets[symbol];
+        symbol = market['id'];
+        const request = {
+            'symbol': symbol,
+            'type': 'kline_' + timeframe,
+        };
+        const response = await this.privateGetKline (this.extend (request, params));
+        return this.parseOHLCVs (response['data'], market, timeframe, since, limit);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        params['apiKey'] = this.apiKey;
```

Comment:
We should not mutate the argument passed from the outside in this case, need to rewrite this as shown, for example, here:
- https://github.com/ccxt/ccxt/blob/master/js/bitso.js#L283
- https://github.com/ccxt/ccxt/blob/master/js/deribit.js#L615

(`params` is always assigned to a different new object, but we don't change the originally-referenced object in the `params` argument)